pkg/buildchange/progress_logger.go

```diff
@@ -0,0 +1,51 @@
+package buildchange
+
+import (
+	"bufio"
+	"context"
+	"io"
+
+	"github.com/acarl005/stripansi"
+
+	corev1 "k8s.io/api/core/v1"
+	k8sclient "k8s.io/client-go/kubernetes"
+)
+
+type ProgressLogger struct {
+	K8sClient k8sclient.Interface
+}
+
+// GetTerminationMessage gets the container logs for a given pod
+func (p *ProgressLogger) GetTerminationMessage(pod *corev1.Pod, s *corev1.ContainerStatus) (string, error) {
+	containerLog, _ := p.getContainerLogs(pod, s)
+
+	moreInfoCmd := createMoreInfoCommand(pod.Namespace, pod.Name, s.Name)
+	return " " + s.State.Terminated.Message + " " + containerLog + moreInfoCmd, nil
```

Comment:
Is there a case where this can get too long that we should account for ?

Reply:
That's a good idea. I've added a truncate method that will cut off at 800 characters. I think this is only relevant for containerLog since the others are mostly predefined ones that predetermined length
- `s.State.Terminated.Message`  is `Container %s terminated with error`
- `moreInfoCmd` is `kubectl logs -n %s %s -c %s`